cg/services/bcl_convert_metrics_service/parser.py

```diff
@@ -175,3 +167,55 @@ def has_undetermined_reads_in_lane(self, lane: int) -> bool:
             metrics=self.quality_metrics, sample_internal_id=UNDETERMINED, lane=lane
         )
         return bool(metrics)
+
+    @classmethod
+    def calculate_total_reads_for_metrics(cls, read_pair_count: int) -> int:
+        """Scale to read pair number up to single reads."""
+        return read_pair_count * SCALE_TO_READ_PAIRS
+
+    def get_aggregate_total_reads_for_metrics(self):
+        """Return the aggregate reads for the whole demux metrics."""
+        aggregate_read_pairs: int = self.get_aggregate_for_attribute(
+            metrics=self.demux_metrics, attr_name="read_pair_count"
+        )
+        return self.calculate_total_reads_for_metrics(read_pair_count=aggregate_read_pairs)
+
+    def get_aggregate_undetermined_reads_for_metrics(self) -> int:
+        """Calculate the total undetermined reads to the demux metrics."""
+        aggregate_undetermined_read_pairs: int = 0
+        for demux_metric in self.demux_metrics:
+            if demux_metric.sample_internal_id == UNDETERMINED:
+                aggregate_undetermined_read_pairs += demux_metric.read_pair_count
+        return self.calculate_total_reads_for_metrics(
+            read_pair_count=aggregate_undetermined_read_pairs
+        )
+
+    def get_aggregate_percent_q30_for_metrics(self) -> float:
+        """Calculate the aggregate percent Q30 for the demux metrics."""
+        aggregate_q30_bases: int = self.get_aggregate_for_attribute(
+            metrics=self.quality_metrics, attr_name="q30_bases_percent"
+        )
+        return round(aggregate_q30_bases / len(self.quality_metrics), 2)
```

Comment:
I think this is a problematic way to aggregate this metric. An aggregate percentage metric will only be the sum of it's part if all parts are the same size. We would need to normalise this somehow. Will come back to this

Reply:
Yes the name is a bit misleading... this is indeed the mean_q30_percent not the aggregate.